### PR TITLE
  ClassLoader.registerAsParallelCapable() must handle null caller

### PR DESCRIPTION
**Pull request by pshipton**
Issue https://github.com/eclipse-openj9/openj9/issues/15168
See also https://github.com/eclipse-openj9/openj9/issues/14600

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/ClassLoader.java [hunk 1]
```diff
@@ -1413,6 +1413,7 @@ private static boolean registerAsParallelCapable(Class<?> callerCls) {
 
 	/*[IF JAVA_SPEC_VERSION >= 19]*/
 	if (!ClassLoader.class.isAssignableFrom(superCls)) {
+		// check the superclass of the caller to only allow subclasses of ClassLoader
 		throw new IllegalCallerException(callerCls.getName());
 	}
 	/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/ClassLoader.java:1419]: I think this check should be done earlier. If the classloader is already registered, a call to this method will succeed no matter who the caller is, contrary to the documentation: https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/ClassLoader.html#registerAsParallelCapable()

Comment (pshipton) [reply jcl/src/java.base/share/classes/java/lang/ClassLoader.java]: I've updated to move the check first to handle a null caller.

**Comment by pshipton** [comment]
jenkins compile amac,xlinux jdk19

**Comment by pshipton** [comment]
Grinder to run java/lang/ClassLoader/BadRegisterAsParallelCapableCaller.java
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/902/

**Comment by pshipton** [comment]
This isn't quite right.

**Comment by pshipton** [comment]
jenkins compile amac,xlinux jdk19

**Comment by pshipton** [comment]
java/lang/ClassLoader/BadRegisterAsParallelCapableCaller.java - passed
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/905/

The jni/nullCaller/NullCallerTest.java doesn't pass because it's got JNI native code which is incompatible with OpenJ9. The first problem is that it's calling FindClass() with an exception pending, which on OpenJ9 return NULL. When that is fixed, the registerAsParallelCapable() part of the test passes. I'll continue to investigate the remaining problems, but meanwhile this PR is ready.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**